### ISSUE TITLE
US-MIDW-MISO: Reporting negative unknown production

### ISSUE THREAD

**Issue by FelixDQ** [tmrowco/electricitymap-contrib#2725]
This seems to happen around ~6 to 11 am UTC every day.

![image](https://user-images.githubusercontent.com/12124251/95666118-0cd0c180-0b57-11eb-8ff6-c8ac7e1426e8.png)

From Kibana:
```
invalid point: {'zoneKey': 'US-MIDW-MISO', 'datetime': datetime.datetime(2020, 10, 9, 23, 30, tzinfo=tzfile('/usr/share/zoneinfo/America/New_York')), 'production': {'coal': 14997.0, 'gas': 16936.0, 'nuclear': 8028.0, 'wind': 13199.0, 'solar': 0.0, 'unknown': -268.0}, 'storage': {}, 'source': 'misoenergy.org', 'schemaVersion': 3}, reason:US-MIDW-MISO: key unknown has negative value -268.0
```

I don't know if this is a bug in the parser or if there something wrong with the data source.

**Comment by thtomate** [comment]
_Other_ contains pumped storage, according to the website and they really have pumped storage (https://en.wikipedia.org/wiki/Ludington_Pumped_Storage_Power_Plant), so negative values can be valid data.  
We just need to move them to storage

**Comment by tth4vb** [comment]
Maybe we can have a check that reclassifies negative production as storage then?

**Comment by systemcatch** [comment]
This came up before in #1482

**Comment by FelixDQ** [comment]
Do we know if this problem is also present in the EIA parser?

It wouldn't be ideal to switch to EIA because then we wouldn't be able to display live data, but it also isn't ideal to have ~5 hours of missing data every day. What are your thoughts?

**Comment by systemcatch** [comment]
I don't see any negative production in the [EIA data](https://www.eia.gov/opendata/qb.php?category=3390150&sdid=EBA.MISO-ALL.NG.OTH.H) for `MISO-Other` however they may count things differently.

**Comment by jarek** [comment]
Would anyone be open to revisiting this issue as per discussion https://github.com/tmrowco/electricitymap-contrib/discussions/3328?

**Comment by robweber** [comment]
Just to cross post here from #3328 I'm curious if any analysis has ever been done of the MISO directly gathered data vs the EIA published data? The `other` category that is causing this trouble is either being integrated or disregarded as part of their dataset. Either it's being reported differently by MISO or EIA directly has some mechanism for handling it if it's not showing up in their datasets. 

I would be curious if cross examining datasets from time periods where negative `other` values are observed would yield an answer about why it's missing in the EIA info? If so could that same mechanism be applied to the MISO parser directly and avoid the 30 hour delay? The data is missing in the EIA data already, why deal with the lag if it could be pulled from MISO and handled in the moment?

**Comment by skovhus** [comment]
@robweber thanks for the interest here and for wanting to improve the map. 

We did some analysis back in January that I'll try to summarize here and list in order of severity:
- We had a lot of missing data ~7% for 2020-Q3 and around 14% summer 2020
- We observed a lot of negative unknown values (see histogram below using January 2021 data). The large portion of large values means we cannot set up a decent validation threshold. I assume the unit is MW.
- We observed that EIA data splits unknown into oil and hydro
- EIA doesn't seem to report pumped hydro storage

![106464205-9e629700-6498-11eb-9911-d938923ba250](https://user-images.githubusercontent.com/1260305/130627571-d2739638-df51-4b9b-92b4-0128a2fde586.png)
